fix(ui): fix refresh data overlap and create button overflow on flows page

Debjyoti2004 opened the pull request:
### ✨ Description

This PR addresses two UI responsiveness issues on the Flows page to improve layout stability on smaller screens while maintaining full functionality on larger monitors.

**Changes:**
1.  **Fix "Refresh data" Overlap (Issue #13406):**
    * Resolved the overlap between the "Refresh data" label and the "Add filters" button.
    * Implemented responsive visibility: The text label is now hidden on standard laptop screens (showing only the icon) and reappears on extra-large screens (`d-xxl-inline`).
    * Added a Tooltip that displays "Refresh data" on hover.
    * Increased the Refresh icon size (18px) for better accessibility when the text is hidden.
    * Added proper spacing (margins/gaps) to the filter bar components.

2.  **Fix Top Toolbar Overflow (Issue #13418):**
    * Prevented the "Create" button from being pushed out of the container.
    * Applied responsive text hiding to secondary actions (Export, Import, Source Search) using `d-xl-inline`.
    * **Smart Tooltips:** Integrated `useBreakpoints` from `@vueuse/core`. Tooltips for these buttons are now logically controlled: they are **enabled** on small screens (when text is hidden) and **disabled** on large screens (when text is visible), preventing redundant UX.
    * The "Create" button text remains always visible as a primary action.

### 🔗 Related Issue

Closes #13406
Closes #13418

### 🎨 Frontend Checklist

- [x] Code builds without errors (`npm run build`)
- [ ] All existing E2E tests pass (`npm run test:e2e`)
- [x] Screenshots or video recordings attached showing the `UI` changes

### 📸 Visuals

| State | Before | After |
| :--- | :--- | :--- |
| **Refresh Button Overlap** | <img width="400" alt="overlapping_Issue" src="https://github.com/user-attachments/assets/f14d2e9b-8549-4858-9850-0a5779ddbe29" /> | <img width="400" alt="overlapping_Issue_after" src="https://github.com/user-attachments/assets/726642c0-b3ce-418f-b766-2baa7c969a0b" /> |
| **Toolbar Overflow** | <img width="400" alt="Create_button_overflows" src="https://github.com/user-attachments/assets/15552392-0e56-4d59-8bd8-c42131a2376e" /> | <img width="400" alt="Create_button_overflows_after" src="https://github.com/user-attachments/assets/5efa7782-a1d4-49f0-abd1-a7c7b189a613" /> |

**Video Demo:**

https://github.com/user-attachments/assets/6436bfb6-65cf-4b11-b26a-1cdc72ddafb0

Debjyoti2004 committed:
fix(ui): resolve overlap between refresh data text and filters button #13406

Debjyoti2004 committed:
fix(ui): prevent create button overflow by making toolbar actions responsive #13418

Piyush-r-bhaskar reviewed (changes requested):
Hey @Debjyoti2004 

do we really need changes in `Flows.vue` if it's only about Refresh data and can you check other pages please ?

Debjyoti2004 commented:
Hi @Piyush-r-bhaskar, thanks for checking this!

To clarify the changes in `Flows.vue`:
These are actually fixing a separate but related UI responsiveness issue (#13418). On smaller screens/laptops, the "Create" button was getting pushed out of the view because the toolbar was too crowded. I implemented similar responsive logic there (hiding text on small screens) to fix the overflow.

So this PR covers both:
1. Refresh Data overlap (#13406) via `RightFilter.vue` / `KSFilter.vue`
2. Toolbar overflow (#13418) via `Flows.vue`

I have checked the main pages where these components are used and they look good. If there is a specific page you are concerned about or would like me to test for side effects, please let me know and I will verify it and share screenshots!

Piyush-r-bhaskar reviewed (changes requested):
Thanks for this @Debjyoti2004 

We don't want to have just a refresh icon it should be as it was earlier on large screen with text +  icon

<img width="2772" height="1892" alt="Image" src="https://github.com/user-attachments/assets/76fc627b-2935-4d50-a0c6-654edbb62b38" />

and please check on `/executions` once you made your changes.

Debjyoti2004 commented:
Hi @Piyush-r-bhaskar,
Just to confirm before I update the code — is this behavior okay?
1. On large screens → the refresh button will show text + icon (same as earlier).
2. On smaller screens → it will collapse to icon-only to avoid overflow.
3. I will also update the hover effect so it matches the design correctly.

Please let me know if this matches what you expect — I’ll make the changes right after your confirmation.

Piyush-r-bhaskar commented:
sounds good , but keep it only till `768px` and then with text . if overflows , wrap it properly.

Debjyoti2004 committed:
fix(ui): update refresh button responsiveness and layout

Debjyoti2004 committed:
Merge branch 'develop' into fix/13406-refresh-data-overlap

Debjyoti2004 commented:
Hi @Piyush-r-bhaskar, thanks for the review.

I have updated the code to reflect the requested changes:
- **Responsive Logic**: The button is now strictly icon-only up to 768px. The text label appears only on screens wider than that.
- **Overflow Handling**: Added white-space: nowrap to ensure the text wraps properly and doesn't break onto a new line if space is tight.
- **UX Improvements**: I fixed the missing cursor: pointer and slightly increased the icon size on mobile for better accessibility.

Verification: Per your request, I tested this on the` /executions` page as well. The component renders correctly there, maintaining the same responsive behavior without breaking the layout.

I've attached screenshots of the behavior on both small and large screens below. Ready for a re-review!
<img width="1710" height="962" alt="Screenshot 2025-12-05 at 7 49 13 PM" src="https://github.com/user-attachments/assets/4397c2ec-b821-41e3-900f-bddaa60642e5" />
<img width="736" height="869" alt="Screenshot 2025-12-05 at 8 06 23 PM" src="https://github.com/user-attachments/assets/669b0d75-600e-42b4-b9ce-45259122a559" />
<img width="1655" height="868" alt="Screenshot 2025-12-05 at 8 07 50 PM" src="https://github.com/user-attachments/assets/a7f2f755-00ee-43d4-8622-59340bb12b60" />
<img width="739" height="864" alt="Screenshot 2025-12-05 at 8 07 37 PM" src="https://github.com/user-attachments/assets/781419cb-9792-4be5-aae8-810db3e574b4" />

MilosPaunovic committed:
Merge branch 'develop' into fix/13406-refresh-data-overlap

Piyush-r-bhaskar commented:
Hey @Nico-Kestra 

Could you give your suggestion how it should be since we had a lot of discussion internally around refresh data button in filter ?  To brief you,  this PR is trying to keep only `icon` for refresh data on `max-width: 768px` and above that `icon + text`

Piyush-r-bhaskar commented:
@Debjyoti2004  I check on mobiles viewport and the refresh data is still overflowing. I discussed with our designer and he will come up with design for `sm` and `md` viewport filters to fix it properly. hence, I am closing this PR.
